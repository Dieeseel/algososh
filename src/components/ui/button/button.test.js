import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { Button } from './button';


afterEach(cleanup);


describe('Тест рендера кнопки', () => {
  it('Рендер кнопки с текстом прошел без ошибок', () => {
    const button = renderer
      .create(<Button text='Текст'/>)
      .toJSON();
      expect(button).toMatchSnapshot();
  });

  it('Рендер кнопки без текста прошел без ошибок', () => {
      const button = renderer
        .create(<Button text='' />)
        .toJSON();
        expect(button).toMatchSnapshot();
  });

  it('Рендер заблокированной кнопки без ошибок', () => {
      const button = renderer
        .create(<Button disabled={true} />)
        .toJSON();
        expect(button).toMatchSnapshot();
  });

  it('Рендер кнопки с индикатором загрузки прошел без ошибок', () => {
      const button = renderer
        .create(<Button isLoader={true} />)
        .toJSON();
        expect(button).toMatchSnapshot();
  });

  it('Колбэк вызывается корректно при нажатии на кнопку', () => {
      const onClick = jest.fn()
      const { getByRole } = render(<Button onClick={onClick} />);

      const button = getByRole('button');
      fireEvent.click(button)
      expect(onClick).toHaveBeenCalled();
  });
})