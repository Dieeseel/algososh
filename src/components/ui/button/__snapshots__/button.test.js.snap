// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест рендера кнопки Рендер заблокированной кнопки без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест рендера кнопки Рендер кнопки без текста прошел без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`Тест рендера кнопки Рендер кнопки с индикатором загрузки прошел без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тест рендера кнопки Рендер кнопки с текстом прошел без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст
  </p>
</button>
`;
